### PR TITLE
chore: Makes it easier to access contents of emitted assets

### LINKED ISSUE #9299: Make it easier to access contents of emitted assets
## Feature request

**What is the expected behavior?**

Prior to `futureEmitAssets` being added in #8642, it was possible to easily access the final contents of all emitted assets at `stats.compilation.assets`.

**What is motivation or use case for adding/changing the behavior?**

This was useful for tracking the size of our bundled assets to avoid unintentional bundle size regressions. While `size` is still exposed on `SizeOnlySource`, it's now harder to get access to the actual contents in order to track gzip/brotli/etc. size. Reading in the final emitted files from the file system would work fine but is harder than it probably should be as various plugins like `MiniCssExtractPlugin` support different public paths for different asset types.

**How should this be implemented in your opinion?**

Including the location of the emitted asset as part of the asset stats would make things easier. Perhaps this is already available somewhere but I've been unable to find it if so.

**Are you willing to work on this yourself?**
Yes.

### PR DESCRIPTION
**Pull request by jamesgeorge007**
Fixes #9299 

**What kind of change does this PR introduce?**

bugfix

**Did you add tests for your changes?**
Nope 

**Does this PR introduce a breaking change?**
Nope

**What needs to be documented once your changes are merged?**
N/A

### CODE REVIEW

File: lib/Compiler.js [hunk 1]
```diff
@@ -56,6 +56,8 @@ class Compiler extends Tapable {
 			emit: new AsyncSeriesHook(["compilation"]),
 			/** @type {AsyncSeriesHook<Compilation>} */
 			afterEmit: new AsyncSeriesHook(["compilation"]),
+			/** @type {AsyncSeriesHook<Compilation>} */
+			assetEmitted: new AsyncSeriesHook(["compilation"]),
```
Comment (jamesgeorge007) [inline lib/Compiler.js]: The `hooks` object was lacking `assetEmitted` key. How should it be refactored :thinking:

**Comment by jamesgeorge007** [comment]
cc @sokra

**Comment by jamesgeorge007** [comment]
cc @sokra

**Comment by jamesgeorge007** [comment]
@sokra I'm not able to figure out what exactly is to be done regarding the unit test. It would be great if you could give me a better picture :clap:

**Comment by jamesgeorge007** [comment]
@sokra I'm not able to figure out what exactly is to be done regarding the unit test. It would be great if you could give me a better picture :clap:

**Comment by sokra** [comment]
Thanks

**Pull request merged**